lib/revert.js

```diff
@@ -44,3 +45,18 @@ Revert.commit = function(
       return result;
     }, callback);
 };
+
+/**
+ * Reverts the given commit, producing changes in the index and
+ * working directory.
+ * 
+ * @async
+ * @param {Repository} repo the repository to perform the revert in
+ * @param {Commit} commit the commit to revert
+ * @param {RevertOptions} revert_options the revert options
+ *                                       (or null for defaults)
+ */
+Revert.revert = function(repo, commit, revertOpts) {
+  revertOpts = normalizeOptions(revertOpts || {}, NodeGit.RevertOptions);
```

Comment:
I believe revert options also has checkout and merge options nested in it. You may need to pull those out and normalize them. Checkout the normalizeFetchOptions.js file to see how we've normalized nested options structures before.